roles/metrics-export.lua

```diff
@@ -221,11 +253,19 @@ local function validate_http(conf)
     end
 
     for i, nodei in ipairs(conf) do
-        local hosti, porti, erri = parse_listen(nodei.listen)
+        local listen_address = nodei.listen
+        if listen_address == nil then
+            listen_address = get_server_from_httpd_role_cfg(nodei.server).listen
+        end
+        local hosti, porti, erri = parse_listen(listen_address)
```

Comment:
We don't need to validate a listen address from the httpd role configuration. Let's httpd role do it.

Reply:
`httpd`'s role validation is not reachable in validate section as we discuss earlier. As I see here, the only way for the `httpd` role to do the validation logic is in the `apply` block.  Is it ok?

Reply:
It should be enough just to check that a server with a name exists in the `roles.httpd` configuration. There is no need to validate the server configuration, it should do the `roles.httpd`.

Reply:
Deleted validation of listen address from the httpd role configuration.